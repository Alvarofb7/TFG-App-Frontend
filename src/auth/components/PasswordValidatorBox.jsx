export const PasswordValidatorBox = ({
	lowerValid,
	upperValid,
	lengthValid,
	numberValid,
	specialCharValid,
	password,
}) => {
	return (
		<main
			className="password-validation-box"
			aria-label="password-validation-box"
			style={{
				display:
					(lengthValid &&
						lowerValid &&
						upperValid &&
						numberValid &&
						specialCharValid) ||
					password === ""
						? "none"
						: "",
			}}
		>
			<div className={lengthValid ? "validated" : "not-validated"}>
				{lengthValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Debe de contener entre 6 y 20 caracteres
			</div>
			<div className={lowerValid ? "validated" : "not-validated"}>
				{lowerValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos una letra minúscula
			</div>
			<div className={upperValid ? "validated" : "not-validated"}>
				{upperValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos una letra mayúscula
			</div>
			<div className={numberValid ? "validated" : "not-validated"}>
				{numberValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos un número
			</div>
			<div className={specialCharValid ? "validated" : "not-validated"}>
				{specialCharValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos un caracter especial
			</div>
		</main>
	);
};
