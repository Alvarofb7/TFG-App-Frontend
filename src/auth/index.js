export * from "./components/PasswordValidatorBox";
export * from "./pages/LoginPage";
export * from "./layout/HomePage";
