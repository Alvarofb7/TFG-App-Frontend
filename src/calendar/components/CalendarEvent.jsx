export const CalendarEvent = ({ event }) => {
	const { title, notes } = event;

	return (
		<span>
			<strong>{title}</strong> {notes && <span>- {notes}</span>}{" "}
		</span>
	);
};
