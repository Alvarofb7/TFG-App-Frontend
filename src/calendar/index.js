
export * from "./components/CalendarEvent";
export * from "./components/CalendarModal";
export * from "./components/FabAddNew";
export * from "./components/FabDelete";
export * from "../ui/components/Navbar";


export * from "./pages/CalendarPage";