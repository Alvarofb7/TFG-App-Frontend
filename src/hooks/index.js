
export * from "./useAuthStore";
export * from "./useCalendarStore";
export * from "./useForm";
export * from "./useNoteStore";
export * from "./useUiStore";