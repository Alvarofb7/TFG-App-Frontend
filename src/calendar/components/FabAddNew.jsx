import { addHours } from "date-fns";
import { useCalendarStore, useUiStore } from "../../hooks";

export const FabAddNew = () => {
	const { openCalendarModal } = useUiStore();
	const { setActiveEvent } = useCalendarStore();

	const handleClick = () => {
		setActiveEvent({
			title: "",
			notes: "",
			allDay: false,
			start: new Date(),
			end: addHours(new Date(), 2),
			user: {},
		});
		openCalendarModal();
	};

	return (
		<button
			type="button"
			className="btn btn-primary fab"
			aria-label="btn-add-event"
			onClick={handleClick}
		>
			<i className="fas fa-plus"></i>
		</button>
	);
};
