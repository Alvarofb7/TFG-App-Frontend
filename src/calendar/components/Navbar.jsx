import { NavLink } from "react-router-dom";
import { useAuthStore } from "../../hooks";

export const Navbar = () => {

    const { user, startLogout } = useAuthStore()

    return (
        <div className="navbar navbar-expand-md navbar-dark bg-dark mb-4 px-4">
            <span className="navbar-brand">
                <i className="fas fa-calendar-alt"></i>
                &nbsp;
                { user.name || "Sin nombre definido" }
            </span>

            <div className="navbar-collapse">
                <div className="navbar-nav">
                    <NavLink
                        className={ ({ isActive }) => `nav-link ${isActive ? "active" : ""}` }
                        to="/calendar"
                    >
                        Calendario
                    </NavLink>

                    <NavLink
                        className={ ({ isActive }) => `nav-link ${isActive ? "active" : ""}` }
                        to="/notes"
                    >
                        Notas
                    </NavLink>
                    <NavLink
                        className={ ({ isActive }) => `nav-link ${isActive ? "active" : ""}` }
                        to="/kanban"
                    >
                        Kanban
                    </NavLink>
                </div>
            </div>

            <button
                className="btn btn-outline-danger"
                onClick={ startLogout }
            >
                <i className="fas fa-sign-out-alt"></i>
                &nbsp;
                <span>Salir</span>
            </button>

        </div>
    )
}
