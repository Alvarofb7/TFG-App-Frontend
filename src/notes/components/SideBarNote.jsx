import { useEffect, useMemo, useState } from "react"
import { useNoteStore } from "../../hooks";

export const SideBarNote = ({ title = "", id, description = "", date, user = {} }) => {

    const { activeNote, setActiveNote, clearActiveNote } = useNoteStore();

    const newTitle = useMemo(() => {
        return title.length > 15
            ? title.substring(0, 15).concat("...")
            : title
    }, [title]);

    const newDescription = useMemo(() => {
        return description.length > 15
            ? description.substring(0, 15).concat("...")
            : description
    }, [description]);

    const setNote = () => {
        if (activeNote?.id === id) {
            clearActiveNote()
        } else {
            setActiveNote({ id, title, description, date, user });
        }
    };

    return (
        <li className="list-group-item card text-center mb-2" key={ id } id={ id } onClick={ setNote }>
            <div className="card-header">
                <p className="card-title">{ newTitle }</p>
            </div>
            <div className="card-body">
                <p className="card-subtitle mb-2 text-body-secondary">{ newDescription }</p>
            </div>
        </li>
    )
}
