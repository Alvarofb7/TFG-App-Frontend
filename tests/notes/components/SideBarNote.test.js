import { fireEvent, render, screen } from "@testing-library/react";
import { useNoteStore } from "../../../src/hooks";
import { SideBarNote } from "../../../src/notes/components/SideBarNote";
import { noteWithId } from "../../fixtures/notesStates";

jest.mock("../../../src/hooks/useNoteStore");

describe("Pruebas en el componente SideBarNote", () => {
	const mockSetActiveNote = jest.fn();
	const mockSetActiveNewNote = jest.fn();

	beforeEach(() => {
		jest.clearAllMocks();
		jest.clearAllTimers();

		useNoteStore.mockReturnValue({
			setActiveNote: mockSetActiveNote,
			setActiveNewNote: mockSetActiveNewNote,
		});
	});

	test("Debe de mostrarse el componente correctamente", () => {
		const { container } = render(
			<SideBarNote key={noteWithId.id} {...noteWithId} />
		);

		const title = screen.getByText(noteWithId.title);

		expect(title).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	test("Debe de llamar a setActiveNewNote al hacer click en una nota y ya esta activa ", () => {
		useNoteStore.mockReturnValue({
			activeNote: noteWithId,
			setActiveNote: mockSetActiveNote,
			setActiveNewNote: mockSetActiveNewNote,
		});

		render(<SideBarNote key={noteWithId.id} {...noteWithId} />);

		const li = screen.getByLabelText(`note-${noteWithId.id}`);
		fireEvent.click(li);

		expect(mockSetActiveNewNote).toHaveBeenCalled();
	});

	test("Debe de llamar a setActiveNote al hacer click en una nota que no estaba activa", () => {
		useNoteStore.mockReturnValue({
			activeNote: null,
			setActiveNote: mockSetActiveNote,
			setActiveNewNote: mockSetActiveNewNote,
		});

		render(<SideBarNote key={noteWithId.id} {...noteWithId} />);

		const li = screen.getByLabelText(`note-${noteWithId.id}`);
		fireEvent.click(li);

		expect(mockSetActiveNote).toHaveBeenCalled();
	});
});
