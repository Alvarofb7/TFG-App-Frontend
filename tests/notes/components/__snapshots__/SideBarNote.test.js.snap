// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente SideBarNote Debe de mostrarse el componente correctamente 1`] = `
<div>
  <li
    aria-label="note-1"
    class="list-group-item mb-1 text-center"
    id="1"
  >
    <div
      class="card-body"
    >
      <p
        class="card-text"
      >
        <b>
          Nota
        </b>
         - 
        1 nov 2023
      </p>
    </div>
  </li>
</div>
`;
